app/models/api_request.rb

```diff
@@ -21,14 +31,30 @@ def host
     raise "Please define a host method in #{self.class}"
   end
 
+  def page_limit
+    "limit=#{PAGE_LIMIT}" if @page
+  end
+
+  def page_offset
+    if @page
+      offset = (@page.to_i - 1) * PAGE_LIMIT
+      "offset=#{offset}"
+    end
+  end
+
   # Inheriting class defines
   def path
     raise "Please define a path method in #{self.class}"
   end
 
   # Inheriting class defines
+  def parameters
+    raise "Please define a parameters method in #{self.class}"
+  end
+
   def query
-    raise "Please define a query method in #{self.class}"
+    req_params = default_params + parameters
+    req_params.compact.join("&")
```

Comment:
[thought] can the query params array be passed directly into either the `URI` classes or `Net::HTTP.start` call? Then those classes could deal with CGI escapes properly. If it can't then:

[suggestion] I am a fan of a single (overridable) method that takes the parameters array and turns it into a useable query string, but we should use the `CGI` class(? module?) to properly escape and join them.